### PR TITLE
[2456, 2466, 2469, 2470] Fixes on reference widget

### LINKED ISSUE #2456: The create element on a widget ref for a mono valued and containment reference isn't working if there is already a value.
* [x] **I have checked that this bug has not yet been reported by someone else**
* [ ] **I have checked that this bug appears on Chrome**
* [x] **I have specified the version** :
- latest

### Steps to reproduce

An example model with a mono valued and containment reference: 

[Studio.zip](https://github.com/eclipse-sirius/sirius-web/files/12846296/Studio.zip)

With a widget reference on this reference, if there is already a value setted, then the create new element action is not working.

### Actual behavior

Nothing append and there is a snackbar with a generic error.

### LINKED ISSUE #2466: [ReferenceWidget] Unable to select in a non containment reference one of my ancestor
The Reference widget prevent the selection of an ancestor of _self_ in a _ReferenceWidget_. This true for the _dropDown_ candidates and for the modal selection. This use case should not be prevented. For example with this use case, we should be able to select an Entity2 ancestor of Entity1 in the linkedTo feature.

image

[ADA_RefWidgetAncestorProblem.zip](https://github.com/eclipse-sirius/sirius-web/files/12890686/ADA_RefWidgetAncestorProblem.zip)
[ADA_RefWidgetAncestorProblemInstance.zip](https://github.com/eclipse-sirius/sirius-web/files/12890687/ADA_RefWidgetAncestorProblemInstance.zip)

### LINKED ISSUE #2469: Issue on reference widget navigation by chips click
By navigating with a reference widget chips value, if the destination element is filter on the explorer, the application crash.

https://github.com/eclipse-sirius/sirius-web/assets/128395825/134e7b63-8530-4113-85f4-9dcf8af87e5a

The error:
`ErrorBoundary caught error TypeError: refDom.current is null`

### LINKED ISSUE #2470: Unable to remove value with the reference widget on mono reference
There is a bug in the `PropertiesWidgetCreationService#handleRemoveValue` methode, in case of mono valued reference, the `eUnset` is perform on the `reference` instead `referenceOwner`.

### LINKED ISSUE #2487: An error can occur in the reference widget transfert modal if the value is not part of its options

### PR DESCRIPTION
**Pull request by frouene**
# Pull request template

## General purpose
What is the main goal of this pull request?
- [x] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: integration-tests/cypress/e2e/project/edit/widget-reference.cy.js [hunk 1]
```diff
@@ -503,4 +410,127 @@ describe('/projects/:projectId/edit - FormDescriptionEditor', () => {
...
+    cy.getByTestId('filterbar-filter-button').click();
+    cy.getByTestId('Entity2').should('not.exist');
+
+    cy.getByTestId('reference-value-Entity2').click();
+    cy.getByTestId('page-tab-Entity2').should('exist');
+    cy.getByTestId('filterbar-close-button').click();
+    cy.getByTestId('selected').findByTestId('Entity2').should('exist');
+  });
+
+  it.only('check widget reference filter ancestor only for containment reference', () => {
```
Comment (pcdavid) [inline integration-tests/cypress/e2e/project/edit/widget-reference.cy.js]: `it.only` should be removed.

File: packages/emf/backend/sirius-components-emf/src/main/resources/messages/sirius-web-emf.properties [hunk 2]
```diff
@@ -9,7 +9,8 @@
...
 #     Obeo - initial API and implementation
-################################################################################################
-INVALID_INPUT=Invalid input type, "{0}" has been received while "{1}" was expected
-UNEXPECTED_ERROR=An unexpected error has occurred, please contact the server administrator
-INVALID_NUMBER=The new value "{0}" is not a valid number
+#################################################################################
+INVALID_INPUT = Invalid input type, "{0}" has been received while "{1}" was expected
+UNEXPECTED_ERROR = An unexpected error has occurred, please contact the server administrator
+INVALID_NUMBER = The new value "{0}" is not a valid number
+UPPER_BOUNDARY_REACHED = Unable to create a new instance of "{0}" in feature "{1}" because it has reach its upper-bound cardinality."
```
Comment (pcdavid) [inline packages/emf/backend/sirius-components-emf/src/main/resources/messages/sirius-web-emf.properties]: The final `"` is not neeed; it shows as part of the message.

File: packages/emf/backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/services/messages/EMFMessageService.java [hunk 3]
```diff
@@ -39,12 +39,17 @@ public String unexpectedError() {
 
     @Override
     public String invalidInput(String expectedInputTypeName, String receivedInputTypeName) {
-        return this.messageSourceAccessor.getMessage("INVALID_INPUT", new Object[] { expectedInputTypeName, receivedInputTypeName });
+        return this.messageSourceAccessor.getMessage("INVALID_INPUT", new Object[]{expectedInputTypeName, receivedInputTypeName});
```
Comment (pcdavid) [inline packages/emf/backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/services/messages/EMFMessageService.java]: The Eclipse formatting profile keeps the spaces between `[`, `{`, etc.

**Review by pcdavid — CHANGES_REQUESTED** [review]

File: integration-tests/cypress/e2e/project/edit/widget-reference.cy.js (hunk 1 above)
Comment (frouene) [reply integration-tests/cypress/e2e/project/edit/widget-reference.cy.js]: Done

File: packages/emf/backend/sirius-components-emf/src/main/resources/messages/sirius-web-emf.properties (hunk 2 above)
Comment (frouene) [reply packages/emf/backend/sirius-components-emf/src/main/resources/messages/sirius-web-emf.properties]: Done

File: packages/emf/backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/services/messages/EMFMessageService.java (hunk 3 above)
Comment (frouene) [reply packages/emf/backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/services/messages/EMFMessageService.java]: Done,
I've changed my format settings

**Review by pcdavid — APPROVED** [review]

**Comment by pcdavid** [comment]
I'll merge once the build is green.

**Pull request merged**